java/KafkaConnectors/README.md

```diff
@@ -32,7 +34,90 @@ Configuration parameters:
 * `source.topic` source topic (default: `source`)
 * `sink.bootstrap.servers` sink cluster bootstrap servers
 * `sink.topic` sink topic (default: `destination`)
+* `sink.transaction.timeout.ms` Sink transaction timeout 
+
+If you are connecting with no-auth and no SSL, above will work. Else you need additional configuration for both source and sink.
+### For IAM Auth
+
+
+* `source.sasl.mechanism` AWS_MSK_IAM
+* `source.sasl.client.callback.handler.class` software.amazon.msk.auth.iam.IAMClientCallbackHandler
+* `source.sasl.jaas.config` "software.amazon.msk.auth.iam.IAMLoginModule required;"
+* `source.security.protocol` SASL_SSL
+* `source.ssl.truststore.location` /usr/lib/jvm/java-11-amazon-corretto/lib/security/cacerts
+* `source.ssl.truststore.password` changeit
+* `sink.sasl.mechanism` AWS_MSK_IAM
+* `sink.sasl.client.callback.handler.class` software.amazon.msk.auth.iam.IAMClientCallbackHandler
+* `sink.sasl.jaas.config` "software.amazon.msk.auth.iam.IAMLoginModule required;"
+* `sink.security.protocol` SASL_SSL
+* `sink.ssl.truststore.location` /usr/lib/jvm/java-11-amazon-corretto/lib/security/cacerts
+* `sink.ssl.truststore.password` changeit
+
 
-### Running locally in IntelliJ
+## Running locally in IntelliJ
 
 To start the Flink job in IntelliJ edit the Run/Debug configuration enabling *'Add dependencies with "provided" scope to the classpath'*.
+
+Provide arguments like following -
+```
+--source.bootstrap.servers localhost:9092 --source.topic source --sink.bootstrap.servers localhost:9092 --sink.topic sink --sink.transaction.timeout.ms 1000
+```
+
+Following is the screenshot of run configuration
+![Run Configuration](images/runConfiguration.png)
+
+## Running locally through Maven command line
+Refer following sample -
+```
+ mvn clean compile exec:java -Dexec.classpathScope="compile" \
+ -Dexec.mainClass="com.amazonaws.services.msf.KafkaStreamingJob" \
+ -Dexec.args="--source.bootstrap.servers localhost:9092 --source.topic source --sink.bootstrap.servers localhost:9092 --sink.topic sink --sink.transaction.timeout.ms 1000" 
+
+```
+
+## Deploying using CloudFormation to Amazon Managed Service for Apache Flink
+This sample assumes that MSK Serverless cluster is created. The flink application routes data ingested in source topic to sink topic without any transformation. 
+
+![Amazon Managed Service for Apache Flink , MSK Serverless example](images/flink-msk-serverless-example.png),
+### Pre-requisite
+1. Create MSK serverless cluster while choosing 3 subnets. Refer https://docs.aws.amazon.com/msk/latest/developerguide/serverless-getting-started.html . 
+2. Once the cluster is created note down subnets ids of the cluster and security group.
+3. Ensure that security group has self referencing ingress rule that allows connection on port 9098. 
+4. You have a user credential using which you can create CloudFormation stack from console or CLI. 
+
```

Comment:
We should explain that, in order to run the application on the development machine, you need to access Kafka from your local machine, either

1. enable public access on MSK (but this is not supported in MSK Serverless)
2. run kafka locally, e.g. in containers

Explaining how to run Kafka in containers is out of scope. But if we want to make this an e2e example it should be e2e also when developing locally
We need at least to mention what is required